cpmpy/transformations/flatten_model.py

```diff
@@ -315,23 +316,35 @@ def get_or_make_var(expr):
     if __is_flat_var(expr):
         return (expr, [])
 
+    if expr_dict is None:
+        expr_dict_getrmake_var = cpm_dict()
```

Comment:
why do you create a custom-name expr_dict here, and not overwrite the one of the arg?
maybe add a line of documentation about it...

Reply:
Ah this is an artifact from trying to dectect where we called __eq__ from, can be reverted